code/modules/antagonists/xeno/xeno.dm

```diff
@@ -31,6 +31,56 @@
 /datum/antagonist/xeno/get_team()
 	return xeno_team
 
+/mob/living/carbon/alien/humanoid/royal/queen/Login()
+	..()
+	SSshuttle.registerHostileEnvironment(src)
+
+/mob/living/carbon/alien/humanoid/royal/queen/Logout()
+	..()
+	addtimer(CALLBACK(src, .proc/logoutcheck), 40 SECONDS)//when logging out, we don't want to spam announcements with leaving and reconnecting.
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
+
+/mob/living/carbon/alien/humanoid/royal/queen/death()
+	..()
+	SSshuttle.clearHostileEnvironment(src)
+	neutralized = TRUE
+
+/mob/living/carbon/alien/humanoid/royal/queen/proc/logoutcheck()
+	if(!client)
+		SSshuttle.clearHostileEnvironment(src) //the queen has officially gone braindead, so we can allow the shuttle to leave
+
+/mob/living/carbon/alien/humanoid/royal/queen/Life()
+	..()
+	if(!client)
+		return
+	if(life_ticks_to_wait)
+		life_ticks_to_wait--
+		return
+	life_ticks_to_wait = initial(life_ticks_to_wait)
+	var/living_humans = 0
+	var/total_humans = length(GLOB.human_list)
+	for(var/H in GLOB.human_list)
+		var/mob/living/carbon/human/human = H
+		if(!human.client || human.stat == DEAD )
+			continue
+		living_humans++
+	if(living_humans < total_humans/10 && !nuking && !neutralized)
+		INVOKE_ASYNC(src, .proc/nuke_it_from_orbit)
+
+/mob/living/carbon/alien/humanoid/royal/queen/proc/nuke_it_from_orbit()
+    nuking = TRUE
+    addtimer(CALLBACK(GLOBAL_PROC, .proc/priority_announce, "Hostile Lifeforms Identified. Extreme Biohazard Alert. Determining Containment Solutions","Central Command Update", 'sound/misc/notice1.ogg'), 50)
+    addtimer(CALLBACK(GLOBAL_PROC, .proc/priority_announce, "Containment Solution Identified. Initiating Station Self Destruct Protocol.","Central Command Update", 'sound/misc/notice1.ogg'), 450)
+    addtimer(CALLBACK(src, .proc/blow_nuke), 500)
+
+/mob/living/carbon/alien/humanoid/royal/queen/proc/blow_nuke()
+    var/obj/machinery/nuclearbomb/selfdestruct/nuke = locate() in GLOB.nuke_list
+    nuke.safety = FALSE
+    nuke.explode()
+
+var/nuking = FALSE
```

Comment:
I don’t think you meant to put this here, and if you did, we already have a global var for if the nuke is going off. Look at `nuclearbomb.dm` for some pointers.

Reply:
Fixed